feat: add how to preview documentation modification (#203)
autowarefoundation/autoware-documentation: docs/contributing/documentation-guidelines/index.md

# Documentation guidelines

> ## Contribution guidelines on Autoware Documentation

Reviewer: What made you add this section? I feel it's a bit redundant.

Author: As I am going to add a new section below, I felt that the table of contents of this page becomes weird without the section name on the part that currently exists. 

With this modification:
![image](https://user-images.githubusercontent.com/44218668/192910391-0e4de612-b2ca-4982-a48f-cca01fa21065.png)

Without:
![Uploading image.png…]()

Reviewer: > With this modification:

Hmm, I still feel weird even with this modification applied.

How about this?

```md
## Workflow

Contributions to Autoware's documentation are welcome ...

## Style guide

You should refer to the ...

## Tips

### How to preview your modification
```

Author: Sorry for the late reply. I will take that idea, thank you!
https://github.com/autowarefoundation/autoware-documentation/pull/203/commits/71cdf56f0afd5c453dcf54d5986d1dc142e54fe6

## Style guide

You should refer to the [Google developer documentation style guide](https://developers.google.com/style) as much as possible. Reading the [Highlights page](https://developers.google.com/style/highlights) of that guide is recommended, but if not then the key points below should be noted.

- [Use standard American English spelling](https://developers.google.com/style/spelling) and punctuation.
- [Use sentence case](https://developers.google.com/style/capitalization) for document titles and section headings.
- [Use descriptive link text](https://developers.google.com/style/link-text).
- [Write short sentences](https://developers.google.com/style/translation#write-short,-clear,-and-precise-sentences) that are easy to understand and translate.

## Tips

### How to preview your modification

There are two ways to preview your modification on a documentation website.

#### 1. Using GitHub Actions workflow

Follow the steps below.

1. Create a pull request to the repository.
2. Add the `documentation` label from the sidebar (See below figure).
3. Wait for a couple of minutes, and the `github-actions` bot will notify the URL for the pull request's preview.

> ![documenatation_label](images/documentation_label_for_pull_request.png){ width="800" }

Reviewer: [nits] Could you use dash-case for files?

```suggestion
![documenatation_label](images/documentation-label-for-pull_request.png){ width="800" }
```

Author: https://github.com/autowarefoundation/autoware-documentation/pull/203/commits/e5d9709efe8d42eb56239c31ac4b6864e521c900
Done!

> ![documenatation_label](images/documentation-label-for-pull-request.png){ width="800" }

Reviewer: typo, and also to be dash-case

```suggestion
![documentation-label](images/documentation-label-for-pull-request.png){ width="800" }
```

Author: https://github.com/autowarefoundation/autoware-documentation/pull/203/commits/f6809facff9c20d66a6b0aee4f67396aba4a45a2
Thanks, done

#### 2. Running an MkDocs server in your local environment

Instead of creating a PR, you can use the `mkdocs` command to build Autoware's documentation websites on your local computer.
Assuming that you are using Ubuntu OS, run the following to install the required libraries.

```bash

> python3 -m pip install -U $(curl -fsSL \
>   https://raw.githubusercontent.com/autowarefoundation/autoware-github-actions/main/deploy-docs/mkdocs-requirements.txt)

Reviewer: [nits] Do you need a `\` here? If you want to make it multiline, I feel it's better to add a `\` before `$(curl`.

```suggestion
python3 -m pip install -U $(curl -fsSL 
 https://raw.githubusercontent.com/autowarefoundation/autoware-github-actions/main/deploy-docs/mkdocs-requirements.txt)
```

Author: https://github.com/autowarefoundation/autoware-documentation/pull/203/commits/88236e298c7fe5349de0703041d0a8630f83872f
Just removed the backslash

- [Use sentence case](https://developers.google.com/style/capitalization) for document titles and section headings.
- [Use descriptive link text](https://developers.google.com/style/link-text).
- [Write short sentences](https://developers.google.com/style/translation#write-short,-clear,-and-precise-sentences) that are easy to understand and translate.

> ## How to preview your modification on Autoware Documentation

Reviewer: Since we have other documentation repositories, would this be better?
https://autowarefoundation.github.io/autoware.universe/main/
https://github.com/autowarefoundation/open-ad-kit-docs

```suggestion
## How to preview your modification on Autoware's documentation repositories
```

Author: Resolved https://github.com/autowarefoundation/autoware-documentation/pull/203/commits/ff49a125b82682e0d062b355a2a4a3258451dcd9

> There are two ways to preview your modification on Autoware Documentation.

Reviewer: The same reason as https://github.com/autowarefoundation/autoware-documentation/pull/203/files#r982397869.

```suggestion
There are two ways to preview your modification on a documentation site.
```

Author: Resolved https://github.com/autowarefoundation/autoware-documentation/pull/203/commits/ff49a125b82682e0d062b355a2a4a3258451dcd9

> ### 1. Use Github Actions workflow in `autowarefoundation/autoware-documentation` repository

Reviewer: ```suggestion
### 1. Use GitHub Actions workflow
```

Author: Resolved https://github.com/autowarefoundation/autoware-documentation/pull/203/commits/ff49a125b82682e0d062b355a2a4a3258451dcd9

> You can use Github Actions in `autowarefoundation/autoware-documentation` repository to deploy your branch.

Reviewer: I feel this sentence can be simplified because it looks like just repeating the previous information.

```suggestion
Follow the steps below.
```

Or I guess you can even drop this sentence.
```suggestion
```

Author: Resolved https://github.com/autowarefoundation/autoware-documentation/pull/203/commits/ff49a125b82682e0d062b355a2a4a3258451dcd9

> 1. Create a PR of your branch to the repository.

Author: Resolved https://github.com/autowarefoundation/autoware-documentation/pull/203/commits/ff49a125b82682e0d062b355a2a4a3258451dcd9

> 2. Add a `documentation` label from the sidebar.

Reviewer: Since there is only one label, I guess `the` is better. (I'm not sure.)
Also, would it be possible for you to add a screenshot of the sidebar? I think that would make it easier to understand.

```suggestion
2. Add the `documentation` label from the sidebar.
```

Author: Resolved https://github.com/autowarefoundation/autoware-documentation/pull/203/commits/ff49a125b82682e0d062b355a2a4a3258451dcd9

> 3. Wait for a couple of minutes, and the `github-actions` will notify the URL for your branch's preview.

Reviewer: - Add `bot` to be specific.
- I feel `pull request` is better than `branch` here.

```suggestion
3. Wait for a couple of minutes, and the `github-actions` bot will notify the URL for your pull request's preview.
```

Author: Resolved https://github.com/autowarefoundation/autoware-documentation/commit/ff49a125b82682e0d062b355a2a4a3258451dcd9

> ### 2. Run mkdocs in your local environment

Reviewer: Looks better? (I'm not sure.)

```suggestion
### 2. Run MkDocs commands in your local environment
```

Author: Resolved https://github.com/autowarefoundation/autoware-documentation/pull/203/commits/ff49a125b82682e0d062b355a2a4a3258451dcd9
I changed it to "Running an MkDocs server in your local environment"

> Instead of creating a PR, you can use `mkdocs` to build the Autoware Documentation website on your local computer.

Reviewer: To be specific.

```suggestion
Instead of creating a PR, you can use the `mkdocs` command to build the Autoware Documentation website on your local computer.
```

Author: Resolved https://github.com/autowarefoundation/autoware-documentation/pull/203/commits/ff49a125b82682e0d062b355a2a4a3258451dcd9

Assuming that you are using Ubuntu OS, run the following to install the required libraries.

> ```
> sudo apt install -y mkdocs
> git clone git@github.com:autowarefoundation/autoware-github-actions.git /tmp/autoware-github-actions
> cd /tmp/autoware-github-actions/deploy-docs
> pip install -U -r mkdocs-requirements.txt
> ```

Reviewer: ``````suggestion
```bash
python3 -m pip install -U $(curl -fsSL https://raw.githubusercontent.com/autowarefoundation/autoware-github-actions/main/deploy-docs/mkdocs-requirements.txt)
```
``````

Author: Resolved https://github.com/autowarefoundation/autoware-documentation/commit/ff49a125b82682e0d062b355a2a4a3258451dcd9

> Then, run `mkdocs serve` on your Autoware Documentation directory.

Reviewer: The same reason as https://github.com/autowarefoundation/autoware-documentation/pull/203/files#r982397869.

```suggestion
Then, run `mkdocs serve` on your documentation directory.
```

Author: Resolved https://github.com/autowarefoundation/autoware-documentation/pull/203/commits/ff49a125b82682e0d062b355a2a4a3258451dcd9

> ```

Author: Resolved https://github.com/autowarefoundation/autoware-documentation/pull/203/commits/ff49a125b82682e0d062b355a2a4a3258451dcd9

cd /PATH/TO/autoware-documentation
mkdocs serve
```

> It will launch mkdocs server. Access [http://127.0.0.1:8000/autoware-documentation/](http://127.0.0.1:8000/autoware-documentation/) to see the preview of Autoware Documentation.

Reviewer: - If it's not the command, I feel using `MkDocs` is better.
- Does it require an article? (I'm not sure.)
- I think you don't need `autoware-documentation` here.

```suggestion
It will launch an MkDocs server. Access [http://127.0.0.1:8000/](http://127.0.0.1:8000/) to see the preview of Autoware Documentation.
```

Author: Resolved https://github.com/autowarefoundation/autoware-documentation/pull/203/commits/ff49a125b82682e0d062b355a2a4a3258451dcd9
I also changed the subsection title as well ("Running an MkDocs server in your local environment")
